### DISCUSSION TITLE
Bring back option to change tray icons from MDL2 to Segoe Fluent Icons

### DISCUSSION THREAD

**Discussion by lixkote** [valinet/ExplorerPatcher#1011] [Ideas]
![image](https://user-images.githubusercontent.com/95425619/156924920-4420deba-6dc4-4ba8-9d25-35ee86a4ea5c.png)

**Comment by lixkote** [comment]
> Yeah, it's possible. I was using some code made mostly from code decompiled from SAB some time ago and it worked.
> 
> ![image](https://user-images.githubusercontent.com/6503598/142724436-44d624c5-4cf3-4521-8b83-e492de2f064a.png)
> 
> The problem is, it's ripped straight from the decompiled binaries... not to mention it needs some refining to be shipped, but it's also questionable whether to ship it. For the moment, it sits in a private header file. There are mentions about it in the source code as well. I don't actively use it at the moment, it's too hard to test with that for 2 slightly different icons basically, and I can live without that just fine. There was some thread discussing this a while ago: #71.
> 
> Other interesting things regarding graphics can be made as well, like replacing the search, widgets and task view buttons icons on the taskbar. This one needs some additional bits to have it ready for shipping as a release, like, grabbing the icons automatically from the "pri" resource files in the system (they are in SVG), and making scaled versions from those using Windows Imaging Component, for example, at runtime, so we avoid shipping copyrighted binaries, but rather use the ones on the user's system, which are shipped by the OS; again, check the source code, there are some mentions about it:
> 
> ![image](https://user-images.githubusercontent.com/6503598/142724638-f4636711-0388-42a3-bfd7-811d409b3b31.png)
> 
> Similarly, maybe even the Start button image could be changed...? Idk, haven't looked on that specifically, but probably with something along these lines (hook `SHCreateStreamOnModuleResource` in `explorer.exe` and provide a stream containing a different icon). The problem, again, is creating scaled versions, so it works properly on different DPI screen etc. Some boilerplate work and quite some support infrastructure around it, that's how graphics work is.
> 
> Anyway, a clean room and tested implementation would be the best regarding this. Otherwise, idk what to really say about this. I personally do not care about graphics that much, so I will leave it up to the community to come up with a production-ready implementation for these things.
> 
> Thanks

**Reply by Poopooracoocoo** [reply]
> so I will leave it up to the community to come up with a production-ready implementation for these things

bruh. the developer already gave you your answer right there.

**Reply by valinet** [reply]
@Lixkote Ask the SAB developer how he feels about this. I can put you in contact with him. Idk, it's a thin line maybe, I mean, it's about respecting each other after all. Not to mention, as @Poopooracoocoo said, I already mentioned this:

> Anyway, a clean room and tested implementation would be the best regarding this. Otherwise, idk what to really say about this. I personally do not care about graphics that much, so I will leave it up to the community to come up with a production-ready implementation for these things.

What's so confusing about it? I haven't heard back from anyone yet. A clean room implementation always can put any debate on matters like this to pasture. But before that, yeah, ask the SAB developer, see what he thinks about this.
